hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/migration/SyntheticCryptoTransferApprovalMigration.java

```diff
@@ -0,0 +1,293 @@
+                cast(null as int) as index,
+                cast(null as bigint) as receiver,
+                account_id as sender,
+                cr.sender_id,
+                token_id,
+                'TOKEN_TRANSFER' as transfer_type
+              from token_transfer t
+              join contract_result cr on cr.consensus_timestamp = t.consensus_timestamp
+              where cr.consensus_timestamp > :lower_bound and
+              cr.consensus_timestamp < :upper_bound and
+              cr.sender_id <> account_id and
+              is_approval = false and
+              t.amount < 0
+            ), nfttransfers as (
+              select (arr.item->>'is_approval')::boolean as is_approval,
+                t.consensus_timestamp,
+                cast(null as bigint) as amount,
+                arr.index,
+                (arr.item->>'receiver_account_id')::bigint as receiver,
+                (arr.item->>'sender_account_id')::bigint as sender,
+                cr.sender_id,
+                cast(null as bigint) as token_id,
+                'NFT_TRANSFER' as transfer_type
+              from transaction t
+              join contract_result cr on cr.consensus_timestamp = t.consensus_timestamp,
+              jsonb_array_elements(nft_transfer) with ordinality arr(item, index)
+              where cr.consensus_timestamp > :lower_bound and
+              cr.consensus_timestamp < :upper_bound and
+              cr.sender_id <> (arr.item->>'sender_account_id')::bigint and
+              (arr.item->>'is_approval')::boolean = false
+            ), entity as (
+                select key, id, timestamp_range from entity
+                where key is not null
+                union all
+                select key, id, timestamp_range from entity_history
+                where key is not null
+            ), cryptoentities as (
+              select ct.*, e.key
+              from cryptotransfers ct
+              join entity e on e.timestamp_range @> ct.consensus_timestamp::bigint
+              where e.id = ct.sender
+            ), tokenentities as (
+              select t.*, e.key
+              from tokentransfers t
+              join entity e on e.timestamp_range @> t.consensus_timestamp::bigint
+              where e.id = t.sender
+            ), nftentities as (
+              select nft.*, e.key
+              from nfttransfers nft
+              join entity e on e.timestamp_range @> nft.consensus_timestamp::bigint
+              where e.id = nft.receiver
+            ) select * from cryptoentities
+              union all
+              select * from tokenentities
+              union all
+              select * from nftentities
+              order by consensus_timestamp asc
+            """;
+
+    private static final String UPDATE_CRYPTO_TRANSFER_SQL =
+            """
+            update crypto_transfer set is_approval = true where amount = :amount and consensus_timestamp = :consensus_timestamp and entity_id = :sender
```

Comment:
crypto transfers in transaction record are aggregated, so remove the amount condition, might as well remove the amount from the select columns in the SQL

```suggestion
            update crypto_transfer set is_approval = true where consensus_timestamp = :consensus_timestamp and entity_id = :sender
```

Reply:
Thanks I have updated the statement and then removed the unneeded amount from the select.